coherence/session.go

```diff
@@ -141,14 +147,34 @@ func NewSession(ctx context.Context, options ...func(session *SessionOptions)) (
 		session.sessOpts.Address = getStringValueFromEnvVarOrDefault(envHostName, "localhost:1408")
 	}
 
-	// if no timeout then use the env or default
-	if session.sessOpts.Timeout == time.Duration(0) {
-		timeoutString := getStringValueFromEnvVarOrDefault(envSessionTimeout, defaultSessionTimeout)
+	// if no request timeout then use the env or default
+	if session.sessOpts.RequestTimeout == time.Duration(0) {
+		timeoutString := getStringValueFromEnvVarOrDefault(envRequestTimeout, defaultRequestTimeout)
 		timeout, err := strconv.ParseInt(timeoutString, 10, 64)
 		if err != nil || timeout <= 0 {
-			return nil, fmt.Errorf("invalid value of %s for timeout", timeoutString)
+			return nil, fmt.Errorf("invalid value of %s for request timeout", timeoutString)
 		}
-		session.sessOpts.Timeout = time.Duration(timeout) * time.Millisecond
+		session.sessOpts.RequestTimeout = time.Duration(timeout) * time.Millisecond
+	}
+
+	// if no disconnect timeout then use the env or default
+	if session.sessOpts.DisconnectTimeout == time.Duration(0) {
```

Comment:
Maybe this should be refactored to avoid duplication?

Reply:
done